builtin/credential/approle/path_role.go

```diff
@@ -867,6 +883,27 @@ func (b *backend) pathRoleRead(req *logical.Request, data *framework.FieldData)
 			resp.AddWarning("Role does not have any constraints set on it. Updates to this role will require a constraint to be set")
 		}
 
+		// People have been complaining about role becoming useless despite
+		// being able to read it. It seems that the reason for that is not
+		// being able to find the role_id to role_name index. For sanity,
+		// verify that the index still exists. If the index is missing, add one
+		// and return a warning so it can be reported.
+		roleIDIndex, err := b.roleIDEntry(req.Storage, role.RoleID)
+		if err != nil {
+			return nil, err
+		}
+		if roleIDIndex == nil {
+			// The index should never be nil for a valid role. If it is, create
+			// a new one.
+			err = b.setRoleIDEntry(req.Storage, role.RoleID, &roleIDStorageEntry{
```

Comment:
Would we need the write lock in this case?

Reply:
Yes, we do! Good catch.